### ISSUE TITLE
proc-macro panic on FromPrimitive for u64

### ISSUE THREAD

**Issue by mlgiraud** [illicitonion/num_enum#102]
The following code
```
use num_enum::FromPrimitive;

fn main() {
    println!("Hello, world!");
}

#[derive(Debug, Eq, PartialEq, FromPrimitive)]
#[repr(u64)]
enum Number {
    Zero = 0,
    #[num_enum(default)]
    NonZero = 1,
}
```
produces the error
```
error: proc-macro derive panicked
 --> src/main.rs:7:32
  |
7 | #[derive(Debug, Eq, PartialEq, FromPrimitive)]
  |                                ^^^^^^^^^^^^^
  |
  = help: message: attempt to shift left with overflow

```

It works as intended when using u32 or smaller types.
Rust version used: stable-x86_64-unknown-linux-gnu - Up to date : 1.67.1 (d5a82bbd2 2023-02-07)

**Comment by mlgiraud** [comment]
@illicitonion Could you please review my fix i provided and merge as soon as you find the time? This is currently a blocker for our project. See #103

**Comment by illicitonion** [comment]
Fixed by #103
